.github/workflows/spread-docs.yaml

```diff
@@ -14,25 +14,70 @@ on:
     - cron: "0 0 * * 0" # Midnight UTC on Sundays
 
 jobs:
+  snap-build:
+    runs-on: [self-hosted, self-hosted-linux-amd64-noble-medium]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Build snap
+        uses: snapcore/action-build@v1
+        id: charmcraft
+      - name: Upload snap artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: snap
+          path: ${{ steps.charmcraft.outputs.snap }}
+
   spread-select:
-    runs-on: spread-installed
+    runs-on: [self-hosted, self-hosted-linux-amd64-noble-medium]
```

Comment:
Can we keep running the select job on a `spread-installed` container? We'll need to keep the addition of the checkout, but then we should be able to skip the setting up of go and spread.